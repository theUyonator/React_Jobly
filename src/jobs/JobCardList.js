import React from "react";
import JobCard from "./JobCard"

/**This component when rendered should show
 * a list of jobs. 
 * 
 * It is used when bothe JobList and CompanyDetail lists 
 * Jobs It receives the function prop which will be called
 * by JobCard on apply
 * 
 * JobList -> JobCardList -> JobCard
 * CompanyDetail -> JobCardList -> JobCard
 */

function JobCardList ({ jobs, apply }) {

    return (
        <div className="JobCardList">
            {jobs.map(job => (
                <JobCard
                    key={job.id}
                    id={job.id}
                    title={job.title}
                    salary={job.salary}
                    equity={job.equity}
                    companyName={job.companyName}
                />
            ))}

        </div>
    );

}


export default JobCardList;