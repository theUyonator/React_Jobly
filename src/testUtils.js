import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
    username: "testuser",
    first_name: "first",
    last_name: "last",
    email: "testuser@test.com",
    photo_url: null,
};

const UserProvider = 
    ({ children, currentuser=demoUser, hasAppliedToJob = () => false }) => (
        <UserContext.Provider value={{ currentuser, hasAppliedToJob}}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };