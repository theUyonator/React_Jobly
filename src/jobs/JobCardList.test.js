import React from "react";
import { render } from "@testing-library/react";
import JobCardList from "./JobCardList";
import { UserProvider } from "../testUtils";

let jobs = [
    {
        id : 1, 
        title: "shoemaker", 
        salary: 1000, 
        equity: 0.5, 
        companyName: "RW & CO"
    }
];

// smoke test
test("renders without crashing", function() {
    render(
    <UserProvider>
        <JobCardList jobs={jobs} />
    </UserProvider>
    );
  });

// snapshot test

test("matches snapshot", function () {
    const { asFragment } = render (
        <UserProvider>
            <JobCardList jobs={jobs}/>
        </UserProvider>
    );
    expect(asFragment()).toMatchSnapshot();
})