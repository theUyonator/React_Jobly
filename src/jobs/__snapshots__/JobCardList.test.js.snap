// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard card"
    >
      <div
        class="card-body"
      >
        <h6
          class="card-title"
        >
          shoemaker
        </h6>
        <p>
          RW & CO
        </p>
        <div>
          <small>
            Salary: 100,0
          </small>
        </div>
        <div>
          <small>
            Equity: 0.5
          </small>
        </div>
        <button
          class="btn btn-danger font-weight-bold text-uppercase float-right"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
